From f24d232950df53820b9cfe45a02d873412e27173 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Wed, 28 Jan 2015 16:32:24 +0800
Subject: [PATCH 0998/1074] usb: gadget: introduce is_selfpowered for
 usb_gadget

commit 80b2502cea34a965a6b3390691854e753945ca5f upstream

Whether the gadget is selfpowerwed or not can be determined by composite
core, so we can use a common entry to indicate if the self-powered
is supported by gadget, and the related private variable at individual
udc driver can be deleted.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Felipe Balbi <balbi@ti.com>
---
 include/linux/usb/gadget.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/include/linux/usb/gadget.h b/include/linux/usb/gadget.h
index 3b17e99..265fdea 100644
--- a/include/linux/usb/gadget.h
+++ b/include/linux/usb/gadget.h
@@ -507,6 +507,7 @@ struct usb_gadget_ops {
  * @name: Identifies the controller hardware type.  Used in diagnostics
  *	and sometimes configuration.
  * @dev: Driver model state for this abstract device.
+ * @is_selfpowered: if the gadget is self-powered.
  * @out_epnum: last used out ep number
  * @in_epnum: last used in ep number
  * @sg_supported: true if we can handle scatter-gather
@@ -555,6 +556,7 @@ struct usb_gadget {
 	struct device			dev;
 	unsigned			out_epnum;
 	unsigned			in_epnum;
+	unsigned			is_selfpowered:1;
 
 	unsigned			sg_supported:1;
 	unsigned			is_otg:1;
-- 
1.7.5.4

